### PR TITLE
Python: Extended Write Example

### PR DESCRIPTION
**Pull request by ax3l**
Add the `7_extended_write_serial` example in python.

### CODE REVIEW

File: src/binding/python/PatchRecordComponent.cpp [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+void init_PatchRecordComponent(py::module &m) {
+    py::class_<PatchRecordComponent, BaseRecordComponent >(m, "Patch_Record_Component")
+        .def("set_unit_SI", &PatchRecordComponent::setUnitSI)
+        .def("reset_dataset", &PatchRecordComponent::resetDataset)
+        .def_property_readonly("ndims", &PatchRecordComponent::getDimensionality)
+        .def_property_readonly("shape", &PatchRecordComponent::getExtent)
+
+        /*
+        template< typename T >
```
Comment (ax3l) [inline src/binding/python/PatchRecordComponent.cpp]: todo: implement

File: examples/7_extended_write_serial.py [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+    f.delete_attribute("custom_attribute_name")
+
+    # setting attributes can be chained in JS-like syntax for compact code
+    tmpItObj = f.iterations[1] \
+        .set_time(42.0) \
+        .set_dt(1.0) \
+        .set_time_unit_SI(1.39e-16)
+    # everything that is accessed with [] should be interpreted as permanent
+    # storage the objects sunk into these locations are deep copies
+    del tmpItObj
```
Comment (anokfireball) [inline examples/7_extended_write_serial.py]: Is this `del` supposed to signal that only the *handle* and not the underlying sunk resource is deleted?
If so, a comment to explain what is going on should be added.

File: examples/7_extended_write_serial.py [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+
+    # this is a reference to an iteration
+    reference = f.iterations[1]
+    reference.set_comment("Modifications to a reference will always be visible"
+                          " in the output")
+    del reference
+
+    # alternatively, a copy may be created and later re-assigned to
+    # f.iterations[1]
+    copy = f.iterations[1]  # TODO .copy()
```
Comment (anokfireball) [inline examples/7_extended_write_serial.py:59]: This is actually outdated in the C++ example. Please add the **TODO** there as well.

File: examples/7_extended_write_serial.py [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+
+    cur_it = f.iterations[1]
+
+    # the underlying concept for numeric data is the openPMD Record
+    # https://github.com/openPMD/openPMD-standard/blob/upcoming-1.0.1/STANDARD.md#scalar-vector-and-tensor-records
+    # Meshes are specialized records
+    cur_it.meshes["generic_2D_field"].set_unit_dimension(
+        {Unit_Dimension.L: -3, Unit_Dimension.M: 1})
+
+    # as this is a reference, it modifies the original resource
```
Comment (anokfireball) [inline examples/7_extended_write_serial.py:75]: Outdated in the C++ example as well.

File: examples/7_extended_write_serial.py [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+    lowRez = cur_it.meshes["generic_2D_field"]
+    lowRez.set_grid_spacing([1, 1, 1, 1, 1, 1]) \
```
Comment (anokfireball) [inline examples/7_extended_write_serial.py]: This `gridSpacing` would actually make it a 6D field.

**Review by anokfireball — APPROVED** [review]
Python example looks good, but C++ needs some love. Will do that when tackling #171.

File: examples/7_extended_write_serial.py (hunk 3 above)
Comment (ax3l) [reply examples/7_extended_write_serial.py:59]: Yes, I thought so. Let's update both.

File: examples/7_extended_write_serial.py (hunk 2 above)
Comment (ax3l) [reply examples/7_extended_write_serial.py]: Let's check this again when updating the C++ example

File: examples/7_extended_write_serial.py (hunk 5 above)
Comment (ax3l) [reply examples/7_extended_write_serial.py]: huh, don't we have the same in the C++ example? Should be three values in both examples, no?

Comment (anokfireball) [reply examples/7_extended_write_serial.py]: The [C++ code](https://github.com/openPMD/openPMD-api/blob/0.6.0-alpha/examples/7_extended_write_serial.cpp#L98) for this reads
```cpp
/* construct a vector of [6, 1] */
lowRez.setGridSpacing(std::vector< double >{6, 1}).setGridGlobalOffset({0, 600});
```
Your interpretation was
```cpp
/* construct a vector of six 1's */
lowRez.setGridSpacing(std::vector< double >(6, 1)).setGridGlobalOffset({0, 600});
```

> Should be three values in both examples, no?

As this is a **2D** mesh, it should contain two values: `[6, 1]`.

https://github.com/openPMD/openPMD-standard/blob/1.1.0/STANDARD.md#required-attributes-for-each-mesh-record

> `gridSpacing`
type: 1-dimensional array containing N *(floatX)* elements, where N is the **number of dimensions in the simulation**

Comment (ax3l) [reply examples/7_extended_write_serial.py]: Oh wow, that's dangerous xD
I thought the initializer-list constructor should read `({6, 1})` but your code is indeed ok and does exactly what you say.

File: examples/7_extended_write_serial.py (hunk 2 above)
Comment (ax3l) [reply examples/7_extended_write_serial.py]: probably not needed anymore. I did not control the garbage collector properly before `0.6.0-alpha`

**Comment by ax3l** [comment]
Ah cheese, I should remove Python 2 builds altogether.

**Pull request merged**